File: easy/0507_perfect_number/solution.py
import math


class Solution:
    def checkPerfectNumber(self, num: int) -> bool:
        if num == 1:
            return False

        divisors = []

        for i in range(1, int(math.sqrt(num)) + 1):
            if num % i == 0:
                divisors.append(i)
                divisors.append(num // i)

        return sum(divisors) - num == num

File: easy/0507_perfect_number/Solution.java
class Solution {
  public boolean checkPerfectNumber(int num) {
    if (num < 2) {
      return false;
    }

    int divisorsSum = 0;

    for (int i = 1; i < (int) Math.sqrt(num) + 1; ++i) {
      if (num % i == 0) {
        divisorsSum += i + num / i;
      }
    }

    return divisorsSum - num == num;
  }
}
